src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -224,6 +224,14 @@ private void InnerReleaseHandle()
             }
         }
 
+        private void TryWakeup(InnerSafeCloseSocket innerSocket)
+        {
+            if ((AsyncContext == null || !AsyncContext.GetNonBlocking()) && innerSocket != null && !_underlyingHandleNonBlocking && !innerSocket.IsClosed && !innerSocket.IsInvalid)
+            {
+                Interop.Sys.Shutdown(innerSocket, SocketShutdown.Receive);
```

Comment:
Just Receive?  Will this fix blocked sends/connects, or do those require shutting down the Send direction as well?

Reply:
I was thinking about it @stephentoub . But I did not want to interfere with writing logic. At least on Unix, close() would always finish writing data. Also the write would finish or fail even if that may take some some. I'm not sure if there is case when it would hang forever. I can revisit that if you think we should.